example/provider-extensions/registry-seed/deploy-registry.sh

```diff
@@ -39,16 +39,95 @@ fi
 kubeconfig=$1
 registry=$2
 
-echo "Generating new password for container registry $registry"
+if kubectl --kubeconfig "$kubeconfig" get secrets -n registry registry-password; then
+  echo "Container registry password found in seed cluster"
+  password=$(kubectl --kubeconfig "$kubeconfig" get secrets -n registry registry-password -o yaml | yq -e .data.password | base64 -d)
+else
+  echo "Generating new password for container registry $registry"
+  password=$(openssl rand -base64 20)
```

Comment:
What is the process to regenerate the password? Do we just need to delete the secret or is it required to do `make gardener-extensions-down gardener-extensions-up`?

Reply:
Please, do not get me wrong. I am aware that this setup is for development and hence may not need the same credential rotation capabilities as a productive setup, but still we probably do not want to use the same password forever...

Reply:
yes, you can rotate the secret.
However, you should avoid to update any gardener image at the same time. When the `node-agent` image tag changes and you rotate the registry password at the same time, all nodes of your shoots will be broken, because `node-agent` cannot pull its own new image anymore.

However, these two ways work:
- run `make gardener-extensions-up` (too ensure that your images are up-to-date)
- reconcile all shoots on the seed where you want to rotate the registry password
- run `kubectl delete secrets -n registry registry-password` on your seed cluster
- run `make gardener-extensions-up`
- reconcile the shoots again

Alternative
- reconcile all shoots on the seed where you want to rotate the registry password
- run `kubectl delete secrets -n registry registry-password` on your seed cluster
- run `./example/provider-extensions/registry-seed/deploy-registry.sh <path to seed kubeconfig> <seed registry hostname>
- reconcile the shoots again

Reply:
I suppose it would be useful to add one of the two approaches as a section in https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally_with_extensions.md.

WDYT?